### PR TITLE
Oauth be

### PR DESCRIPTION
**Pull request by IamNorma**
- Add rack-cors gem
- Update User model with ::from_omniauth() which searches for user by uid and if user does not already exist in database, new user will be created
- Updates user strong params which needs to be cleaned up, but it works for now

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -7,4 +7,14 @@ class User < ApplicationRecord
...
+
+  def self.from_onmiauth(user_params)
+    user = find_or_create_by(uid: user_params[:uid])
+    user.email = user_params[:email]
+    user.provider = user_params[:provider]
+    user.token = user_params[:token]
+    user.refresh_token = user_params[:refresh_token]
+    user.save
+    user
+  end
```
Comment (hashmaster3k) [inline app/models/user.rb:19]: This is cool. So it creates or pulls up a user and basically updates their information from Google? I like it.

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -28,6 +28,10 @@ def destroy
   private
 
   def user_params
-    params.permit(:email, :provider, :token, :refresh_token, :uid)
+    # needs to be cleaned up
+    a = params.permit(:uid, :provider)
+    b = params.require(:credentials).permit(:token, :refresh_token)
+    c = params.require(:info).permit(:email)
+    a.merge(b).merge(c)
```
Comment (hashmaster3k) [inline app/controllers/api/v1/users_controller.rb:35]: I know it's not the best way of going about it but I'm glad it worked!

**Review by hashmaster3k — CHANGES_REQUESTED** [review]
Have a test that is failing, other than that looks good.

**Review by dcoleman21 — COMMENTED** [review]
Looked over this with Norma. @hashmaster3k if you want to sync back up with Norma later, please do.

**Pull request merged**